### ISSUE TITLE
str_c() could be included in unneeded_concatenation_linter()

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1787]
Have seen a few usages like `str_c("a")`, similar to how I used to see `c("a")` which was caught by `unneeded_concatenation_linter()`.

**Comment by IndrajeetPatil** [comment]
Only usages of `str_c()` with `character` type arguments should be linted, right?

If not, the lint message will need to be different because `str_c()` always returns a character.

``` r
library(stringr)

c(1L)
#> [1] 1
str_c(1L)
#> [1] "1"

c()
#> NULL
str_c()
#> character(0)
```

<sup>Created on 2022-12-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
good point... `literal_coercion_linter()` should apply in that case, right?
